seqr/utils/file_utils.py

```diff
@@ -64,3 +64,25 @@ def _google_bucket_file_iter(gs_path, byte_range=None, raw_content=False, user=N
             line = line.decode('utf-8')
         yield line
 
+
+def get_vcf_filename(vcf_path):
+    if vcf_path.endswith('.vcf') or vcf_path.endswith('.vcf.gz') or vcf_path.endswith('.vcf.bgz'):
+        return vcf_path
+    process = subprocess.Popen('gsutil ls ' + vcf_path, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
+    if process.wait() != 0:
+        return None
+    for line in process.stdout:
+        file = line.decode('utf-8').strip()
+        if file.endswith('.vcf') or file.endswith('.vcf.gz') or file.endswith('.vcf.bgz'):
+            return file
+    return None
+
+
+def get_vcf_samples(vcf_path):
+    vcf_filename = get_vcf_filename(vcf_path)
+    if not vcf_filename:
+        return {}
+    for line in file_iter(vcf_filename, byte_range=(0, 65536)):
```

Comment:
`65536` here is the max block size for BGZIP. It ensures the data downloaded include a completed gzip block.

Reply:
We should use a constant for the hardcoded number with a comment explaining this

Also, this helper function doesn't belong in `file_utils` which includes really generic functions for interacting with files, it just belongs in anvil_workspace_api.py . If you wanted to do a generic helper function like `get_google_bucket_file_header` that could live in this file

Reply:
Also, if the file is NOT gzipped, can we be confident that that will be enough bytes for the whole header?

Reply:
I would probably move the functions to `anvil_workspace_api.py`.

For non-gzipped files, I tried 1000 genome project data which contains 2500+ samples, and found that the header is within 64k bytes. I could add a loop to read more blocks. But I'm sure if it is overkilled or not.

Reply:
I would check if a file is gzipped and only include the range size if it is. If its not gzipped, it will only read one line at a time which shouldn't be an issue in terms of size

Reply:
The VCFs are on Google cloud. Is there any way to read them line by line?

Reply:
`file_iter` reads them line by line already

Reply:
alternatively, we could just require vcfs to be gzipped. By default the GATK joint calling workflow in AnVIL gzips things and theres no reason *not* to store a smaller version of the file